packages/bot-web-ui/src/components/dashboard/bot-stopped.spec.tsx

```diff
@@ -0,0 +1,96 @@
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import BotStopped from './bot-stopped';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+jest.mock('@deriv/deriv-charts', () => ({
+    setSmartChartsPublicPath: jest.fn(),
+}));
+
+describe('<BotStopped />', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeAll(() => {
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    const originalLocation = global.location;
+    delete global.location;
+    global.location = {
+        ...originalLocation,
+        replace: jest.fn(),
+        reload: jest.fn(),
+    };
+
+    it('renders the BotStopped component', () => {
+        mock_DBot_store?.dashboard?.setWebSocketState(false);
+
+        render(<BotStopped />, {
+            wrapper,
+        });
+
+        expect(screen.getByText('You’re back online')).toBeInTheDocument();
+    });
+    it('should go to reports page on click of go to reports button', async () => {
+        mock_DBot_store?.dashboard?.setWebSocketState(false);
+
+        render(<BotStopped />, {
+            wrapper,
+        });
+        act(() => {
+            screen.getByText('Go to Reports').click();
+        });
+        await waitFor(() => {
+            expect(global.location.replace).toHaveBeenCalledWith('reports/positions');
+        });
+    });
+
+    it('should reload page on click of Back to Bot', async () => {
+        mock_DBot_store?.dashboard?.setWebSocketState(false);
+
+        render(<BotStopped />, {
+            wrapper,
+        });
+        act(() => {
+            screen.getByText('Back to Bot').click();
+        });
+        await waitFor(() => {
+            expect(global.location.reload).toHaveBeenCalled();
+        });
+    });
+
+    it('should reload page on click of x button', async () => {
+        mock_DBot_store?.dashboard?.setWebSocketState(false);
+
+        render(<BotStopped />, {
+            wrapper,
+        });
+        act(() => {
+            // eslint-disable-next-line testing-library/no-node-access
+            const svgCloseIcon = document.querySelector('.dc-dialog__header--close');
```

Comment:
I am not sure but maybe there is no data-testid in the dc-dialog. If so then it's fine else if there is a data-testid then can we just try to use that instead of directly accessing the node with the class?

Reply:
This component is coming from the shared package and so could not add a data test-id there. So to prevent adding changes in there in the shared package used queryselector.